app/services/google/search_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Google
+  module SearchService
+    SearchServiceError = GoogleScraperRuby::Errors::SearchServiceError
+    HOST = 'www.google.com'
+    # rubocop:disable Layout/LineLength
+    USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36'
+    # rubocop:enable Layout/LineLength
+
+    def self.search!(keyword)
+      uri = URI::HTTPS.build(host: HOST, path: '/search', query: { q: keyword }.to_query)
+      conn = build_conn uri
+      response = conn.get(uri, {}, { 'User-Agent' => USER_AGENT })
+      if !response.success?
+        raise SearchServiceError.new(uri,
+                                     "Response with status code #{response.status}")
+      end
+
+      response.body
+    rescue Faraday::ConnectionFailed, Faraday::ServerError, Faraday::ClientError => e
+      raise SearchServiceError.new uri, e
+    end
+
+    def self.build_conn(uri)
```

Comment:
`conn` is something taken from Phoenix which has no place in Rails 😅 Here the method name should be:

```suggestion
    def http_client
```

Since it is not used elsewhere, it could be a private instance method in a class.

Reply:
actually, it's Faraday's terminology (https://www.rubydoc.info/github/lostisland/faraday/Faraday.new)

However, I admit that a framework-agnostic name would be better. Good point.

Reply:
Fixed in https://github.com/Goose97/google-scraper-ruby/pull/20/commits/2f3c3930c6f5b0f8a32041d9e8e88ae9b7e91828